### ISSUE TITLE
Update to latest bindgen and fix everything

### ISSUE THREAD

**Issue by nlfiedler** [nlfiedler/magick-rust#46]
The most recent bindgen release (0.42.0) causes many compiler warnings and errors with magick-rust. These are probably pretty easy to fix, so now's a good time to take care of that, before we fall too far behind to be manageable.

**Comment by nlfiedler** [comment]
Unfortunately, the docker rust image fails to build because it cannot find `libclang`.

**Comment by nlfiedler** [comment]
Version 0.31 is okay, all versions between that and 0.42 cause bizarre compiler errors in the docker rust image, while 0.42 has the libclang path problem.

**Comment by nlfiedler** [comment]
Actually a bug in clang-sys crate: 
https://github.com/KyleMayes/clang-sys/issues/83

**Comment by nlfiedler** [comment]
Tried 0.44 and that works on macOS but fails with E0204 on Linux (docker image).

**Comment by nlfiedler** [comment]
Tried version 0.51.0: works fine on macOS and fails on the Linux docker image:
```
the trait `std::default::Default` is not implemented for `[u8; 44]
```
Will keep waiting for a version that works.

**Comment by nlfiedler** [comment]
Also tried Ubuntu 18.04 and that resulted in the same compile failure. Probably a general Linux issue that rust-bindgen has yet to deal with.

**Comment by npajkovsky** [comment]
I have fixed that issue and will submit pullreq in couple hours.

**Comment by npajkovsky** [comment]
pullreq submitted.

**Comment by nlfiedler** [comment]
Fixed in commit 92b12900ab0f90d66e455ff00e1edbe724c83f6d
